pylint/lint/report_functions.py

```diff
@@ -38,10 +38,10 @@ def report_messages_stats(
         if not msg_id.startswith("I")
     )
     in_order.reverse()
-    lines = ["message id", "occurrences"]
+    lines: List[str] = ["message id", "occurrences"]
```

Comment:
I think this should be implicit. Marc commented on my previous PR about this, so I think he would like to leave implicit typing. See https://github.com/PyCQA/pylint/pull/4973#discussion_r703074786 and other comments in his earlier reviews.

Reply:
`List[str]` is just not really necessary.

Reply:
Removed in latest rebase.